pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -498,6 +495,97 @@ func (r *ReconcileNetworkAddonsConfig) stopTrackingObjects() {
 	r.statusManager.SetFromPods()
 }
 
+func updateObjectstLabels(crLables map[string]string, objs []*unstructured.Unstructured) ([]*unstructured.Unstructured, error) {
+	var err error
+	for _, obj := range objs {
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		// Label objects with version of the operator they were created by
+		labels[cnaov1.SchemeGroupVersion.Group+"/version"] = operatorVersionLabel
+
+		labels, err = updateLabels(obj, labels, crLables)
+		if err != nil {
+			return objs, err
+		}
+
+		labels, err = updateManagedByLabel(obj, labels, crLables)
+		if err != nil {
+			return objs, err
+		}
+
+		obj.SetLabels(labels)
+	}
+
+	return objs, nil
+}
+
+func updateLabels(obj *unstructured.Unstructured, labels, crLables map[string]string) (map[string]string, error) {
+	var err error
+	const partOfLabelKey = "app.kubernetes.io/part-of"
+	const versionLabelKey = "app.kubernetes.io/version"
+	for key, label := range crLables {
```

Comment:
I think it's better to not iterate over all the crLabels.
Since it's a `map[string]string`, maybe it's better to do something like
```
appLabelKeys := []string{partOfLabelKey, versionLabelKey}
	for _, key := range appLabelKeys {
		if value, exist := crLables[key]; exist == true {
			labels, err = updateLabelByKey(obj, key, value, labels)
			if err != nil {
				return labels, err
			}
		}
	}
```